### ISSUE TITLE
Searching by GPS coordinates not working

### ISSUE THREAD

**Issue by pogos-ed** [organicmaps/organicmaps#6505]
Hi. 
Some regions use "," sign to separate decimals, some regions use "." - this can be set up in Android language settings.

Searching in Organic Maps by coordinate like {41.90195763115396, 12.453785971969468} works fine, but if try to search by coordinate like {41,90195763115396, 12,453785971969468} it doesn't show you the correct point. 

**Steps to reproduce**
1. Set up in android language settings "," as decimals separator
2. Search something in Google Maps
3. Copy gps coordinates from Google Maps and paste to Organic Maps

**Comment by biodranik** [comment]
@vng as we already have an option to retrieve the decimal separator from the locale, let's also use it in the search. It can be done only after pasting something, or also when a user types something.
A possible implementation may replace all locale separators with a dot before passing a search query to the engine.

**Comment by pastk** [comment]
It would be good to support a `42.0778/24.6744` format/separator - this is how it looks in osm.org's urls (e.g. `https://www.openstreetmap.org/#map=13/42.0778/24.6444`) and I copy coords from there often :)

**Comment by lafrech** [comment]
I've just been sent a Google Maps shortened link. When I open it in a browser, I shortly get the coordinates displayed as `42.0778,24.6744` but as soon as the map is displayed, the place is written as `42°07'78.2''N 24°67'44.2W`. I don't know a simple way to find the location in Organic Maps.

Understanding that last form as well would be nice.

Not sure that makes sense here, but perhaps every time the string can be coerced into a GPS coords pair, it should be tried first. This could be done aggressively if the user can add quotes around the string to prevent it for the rare cases where it would be wrong.

**Comment by biodranik** [comment]
Can you please share a link?

**Comment by lafrech** [comment]
I believe the link I was sent was generated by the "share" feature of Google maps: https://maps.app.goo.gl/nCtUdk1diaZmdbUX8. When clicked, it opens this page : https://www.google.fr/maps/place/31%C2%B030'30.1%22N+34%C2%B028'03.6%22E/@31.50837,34.467671,17z/data=!3m1!4b1!4m4!3m3!8m2!3d31.50837!4d34.467671?entry=ttu.

In a browser on my desktop, I can see `31.508370, 34.467671` in the search bar and both `31°30'30.1"N 34°28'03.6"E` and `31.508370, 34.467671` in the place name. But on my mobile (Fennec browser on LineageOS 18, FWIW), I only see the `31°30'30.1"N 34°28'03.6"E` part.

It would be nice if Organic Maps could swallow any of those forms

- `31.508370, 34.467671`
- `31°30'30.1"N 34°28'03.6"E`
- the URL itself: `https://www.google.fr/maps/place/31%C2%B030'30.1%22N+34%C2%B028'03.6%22E/@31.50837,34.467671,17z/data=!3m1!4b1!4m4!3m3!8m2!3d31.50837!4d34.467671?entry=ttu`
- perhaps also OSM URLs (I haven't tried yet)

the point being to avoid unpractical select/copy/paste.

(I'm mentioning the shortened link but its resolution can only be achieved by Google and I'd expect at best Organic Maps to parse the resolved URL, of course.)

If the link was generated from a place rather than a random point on the map, it may look like this: `https://maps.app.goo.gl/TGsVaZcGKJmDNPX78` which points to `https://www.google.fr/maps/place/Gaza/@31.5016946,34.4668445,15z/data=!3m1!4b1!4m6!3m5!1s0x14fd7f054e542767:0x7ff98dc913046392!8m2!3d31.5016951!4d34.4668445!16zL20vMGZka2w?entry=ttu`. In this case, there are no coordinates displayed but only the place name so the coordinates can only be parsed from the URL.

In a webapp I'm developping, I added a feature to allow users to paste Google Maps coords because this is most likely from there they'll be getting the coords in the first place.

```py
@blp.get("/api/googlemaps2coords")
def googlemaps2coords():
    url = flask.request.args.get("url") or flask.abort(422)
    coords = url.partition("@")[2].split(",")[:2]
    if len(coords) != 2:
        flask.abort(422)
    return flask.jsonify(
        {
            "latitude": coords[0],
            "longitude": coords[1],
        }
    )
```

I'm kinda new to Organic Maps and even the mobile/Android world. Please forgive me if this is totally out of scope.

**Comment by biodranik** [comment]
@rtsisyk almost finished a privacy proxy to hide the user's IP from google for such URLs.
